### ISSUE TITLE
Problem with content menu item content picker

### ISSUE THREAD

**Issue by rjpowers10** [OrchardCMS/OrchardCore#7273]
EDIT: OC version is 1.0.0-rc2-14774
EDIT 2: I forgot to mention that this had been working at one point, back when I initially created my menu some number of weeks ago. It was only yesterday that I noticed this issue when I went to add a new menu item.

I'm not sure if this is a bug or something that's gone wrong in my test site, but I haven't been able to solve the problem and I'm unfamiliar with Vue.

Basically, there seems to be a problem with the content picker field when I create a new content menu item. All I see is "not published" for each item.

![image](https://user-images.githubusercontent.com/68876423/96001611-382d0800-0e06-11eb-89df-cdde2d1ab01b.png)

I've debugged DefaultContentPickerResultProvider and confirmed that it is returning the results that I expect, and that everything in the result list has been published (ContentPickerResult.HasPublished = true).

I've also debugged vue-multiselect-wrapper.js, and it seems like I'm getting the correct JSON (again, everything has HasPublished set to true).

But for some reason, the dropdown doesn't seem to be rendering correctly.

I tried a fresh OC site using the blog recipe, and it seems to be working just fine there.

![image](https://user-images.githubusercontent.com/68876423/96002117-c43f2f80-0e06-11eb-9b2e-c64e15a26df8.png)

I'm just looking for any tips on how to figure out what is going on. Thanks!

**Comment by jptissot** [comment]
Which version of OC are you using ?

**Comment by deanmarcussen** [comment]
Are you setting a `DisplayText` value for these content items, i.e. using the `TitlePart` ?

That looks like something you might see if you had no `DisplayText` value

**Comment by rjpowers10** [comment]
Yeah, I'm using a `TitlePart`.

I forgot to mention that this had been working at one point, back when I initially created my menu some number of weeks ago. It was only yesterday that I noticed this issue when I went to add a new menu item.

**Comment by Skrypt** [comment]
Go to your Lucene Indexes and rebuild them.

**Comment by rjpowers10** [comment]
> Go to your Lucene Indexes and rebuild them.

Nope, that didn't have any affect either. I only have one, "Search", and rebuilding it made no difference.

**Comment by Skrypt** [comment]
Can you go in your database and check if the ContentItemIndex.DisplayText column has any data?

**Comment by rjpowers10** [comment]
> Can you go in your database and check if the ContentItemIndex.DisplayText column has any data for this specific ContentType?

Yes. As a test I limited the content picker to a single content type and I verified that ContentItemIndex.DisplayText is populated for all content items of that content type.

What may or may not be interesting is that the dropdown does have the correct number of options (I happen to have three items of this content type, and I have three dropdown options), so at least some part of it is working.

**Comment by Skrypt** [comment]
One of the reasons why it could not work is that this DisplayText gets changed on a EventHandler. Though, I think I'd need to be able to repro locally or else we need to debug this on your solution.

**Comment by rjpowers10** [comment]
In vue-multiselect-wrapper.js I'm getting results that look good to me. But still, I see no results in the content picker.

![image](https://user-images.githubusercontent.com/68876423/96784117-ac921900-13bb-11eb-8adc-ebfc5e1702a1.png)

![image](https://user-images.githubusercontent.com/68876423/96784190-b1ef6380-13bb-11eb-9a53-68c831f2a773.png)

I don't know how to follow the trail from here. I assume that after vm.options is set, Vue takes over. I also don't see any exceptions in the console.

**Comment by jptissot** [comment]
I am unable to replicate this locally. Is there a way for you to share your site / code so we could replicate this issue ?

**Comment by rjpowers10** [comment]
Copied from my PR:

Oh no! I've been betrayed by one of my own team members...look at what I found in our code:

```
services.AddControllers()
    .AddXmlSerializerFormatters()
    .AddNewtonsoftJson(options =>
    {
        options.UseMemberCasing();
    });
```

I removed this and everything started working. As for my PR, I'll don't feel strongly about it either way. I could maybe see an argument for explicitly using camel case here, to avoid problems like this in the future, but I also won't feel bad it you just want to call this "user error" and drop the PR.

I'm sorry for wasting everyone's time on this, but I really appreciate the help!
